src/test/vat.t.sol

```diff
@@ -676,6 +676,63 @@ contract BiteTest is DSTest {
         assertEq(vat.balanceOf(address(vow)),  75 ether);
     }
 
+    // tests a partial lot liquidation because it would fill the literbox
+    function test_partial_litterbox_realistic_values() public {
+        // spot = tag / (par . mat)
+        // tag=5, mat=2
+        vat.file("gold", 'spot', ray(2.5 ether));
+        vat.frob("gold", me, me, me, 100 ether, 150 ether);
+
+        // tag=4, mat=2
+        vat.file("gold", 'spot', ray(1 ether));  // now unsafe
+        cat.file("gold", "chop", 1.13 ether);
+
+        assertEq(ink("gold", address(this)), 100 ether);
+        assertEq(art("gold", address(this)), 150 ether);
+        assertEq(vow.Woe(), 0 ether);
+        assertEq(gem("gold", address(this)), 900 ether);
+
+        cat.file("box", rad(75 ether));
+        cat.file("gold", "dunk", rad(100 ether));
+        assertEq(cat.box(), rad(75 ether));
+        assertEq(cat.litter(), 0);
+        uint auction = cat.bite("gold", address(this));
+        (,,,,,,,uint tab) = flip.bids(auction);
+
+        assertTrue(cat.box() - cat.litter() < ray(1 ether)); // Rounding error to fill box
+        assertEq(cat.litter(), tab);                         // tab = 74.9999... RAD
+
+        uint256 dart = rad(75 ether) * WAD / RAY / 1.13 ether; // room / rate / chop
+        uint256 dink = 100 ether * dart / 150 ether;
+
+        assertEq(ink("gold", address(this)), 100 ether - dink); // Taken in vat.grab
+        assertEq(art("gold", address(this)), 150 ether - dart); // Taken in vat.grab
+        assertEq(vow.sin(now), rad(dart));                      // dart * rate
+        assertEq(gem("gold", address(this)), 900 ether);
+
+        assertEq(vat.balanceOf(address(this)), 150 ether);
+        assertEq(vat.balanceOf(address(vow)),    0 ether);
```

Comment:
Here and elsewhere: why `vat.balanceOf` instead of just `vat.dai`? The latter is arguably more immediately clear in its intent.

Reply:
I guess you aren't the one who started this convention, but I personally just find it annoying to use a wrapper for testing when there's a perfectly good getter already.

Reply:
Yeah I was following the convention which is converting `vat.dai` to a wad: 
```
contract TestVat is Vat {
    uint256 constant ONE = 10 ** 27;
    function mint(address usr, uint wad) public {
        dai[usr] += wad * ONE;
        debt += wad * ONE;
    }
    function balanceOf(address usr) public view returns (uint) {
        return dai[usr] / ONE;
    }
}
```
But for this test especially we should keep it in rad. Good catch I'll switch it to `vat.dai`